pylabrobot/resources/resource.py

```diff
@@ -136,16 +136,34 @@ def assign_child_resource(self, resource: Resource, location: Optional[Coordinat
     which case you will be responsible for handling any checking, if necessary.
     """
 
-    resource.parent = self
-    resource.location = location
     try:
+      self.check_assignment(resource, location)
```

Comment:
`check_assignment` is sufficiently short so that we can put it above this try block. Happy to discuss.

Reply:
Sure, I only wanted it above this, to catch the errors before modifying the objects.

```
      resource.parent = self
      resource.location = location
```

I put it into the try clause just to reuse it, not a strong preference.

Im not confident about the place. Perhaps you can choose a better one?

Perhaps inside the callback? Which seems to do a name check already.

Reply:
I think it's fine to put it above the assignment of `location` and `parent`, as the first code in this method. That way we do not need set the values to `None` if an error occurs.

The reason I assigned them before calling the callback (and unassign them on error), is that some callbacks will want to use this information do determine validity. That is not the case here.

Reply:
Done :ok: